containers/src/Utils/Containers/Internal/EqOrdUtil.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE CPP #-}
+module Utils.Containers.Internal.EqOrdUtil
+  ( EqM(..)
+  , OrdM(..)
+  ) where
+
+#if !MIN_VERSION_base(4,11,0)
+import Data.Semigroup (Semigroup(..))
+#endif
+import Utils.Containers.Internal.StrictPair
+
+newtype EqM a = EqM { runEqM :: a -> StrictPair Bool a }
```

Comment:
What is this?

Reply:
This exists for its semigroup instance.

As documented, it composes the functions and short-circuits on false. This is used in the Eq implementations where false means a pair of unequal elements were found, and due to short-circuiting it skips further comparisons.

Reply:
I still don't get the point. Why not just `foldrWithKey` over the `Map`, passing the `Stack` along?

Reply:
Ah yes, that turns out to be slower. Just replied with it in the comment below.